### ISSUE TITLE
intro-0.5.2.0 testsuite fails to build in Stackage Nightly

### ISSUE THREAD

**Issue by juhp** [minad/intro#2]
In the Stackage Nightly build:
```
Building test suite 'test' for intro-0.5.2.0..
[1 of 4] Compiling BaseCompat       ( test/BaseCompat.hs, dist/build/test/test-tmp/BaseCompat.o )

test/BaseCompat.hs:4:3: error:
    Conflicting exports for ‘cycle’:
       ‘module X’ exports ‘X.cycle’
         imported from ‘Intro’ at test/BaseCompat.hs:7:1-17
         (and originally defined in ‘Data.List.NonEmpty’)
       ‘module X’ exports ‘X.cycle’
         imported from ‘Prelude’ at test/BaseCompat.hs:61:1-178
         (and originally defined in ‘GHC.List’)
  |
4 |   module X
  |   ^^^^^^^^

test/BaseCompat.hs:4:3: error:
    Conflicting exports for ‘head’:
       ‘module X’ exports ‘X.head’
         imported from ‘Intro’ at test/BaseCompat.hs:7:1-17
         (and originally defined in ‘Data.List.NonEmpty’)
       ‘module X’ exports ‘X.head’
         imported from ‘Prelude’ at test/BaseCompat.hs:61:1-178
         (and originally defined in ‘GHC.List’)
  |
4 |   module X
  |   ^^^^^^^^

test/BaseCompat.hs:4:3: error:
    Conflicting exports for ‘init’:
       ‘module X’ exports ‘X.init’
         imported from ‘Intro’ at test/BaseCompat.hs:7:1-17
         (and originally defined in ‘Data.List.NonEmpty’)
       ‘module X’ exports ‘X.init’
         imported from ‘Prelude’ at test/BaseCompat.hs:61:1-178
         (and originally defined in ‘GHC.List’)
  |
4 |   module X
  |   ^^^^^^^^

test/BaseCompat.hs:4:3: error:
    Conflicting exports for ‘last’:
       ‘module X’ exports ‘X.last’
         imported from ‘Intro’ at test/BaseCompat.hs:7:1-17
         (and originally defined in ‘Data.List.NonEmpty’)
       ‘module X’ exports ‘X.last’
         imported from ‘Prelude’ at test/BaseCompat.hs:61:1-178
         (and originally defined in ‘GHC.List’)
  |
4 |   module X
  |   ^^^^^^^^

test/BaseCompat.hs:4:3: error:
    Conflicting exports for ‘tail’:
       ‘module X’ exports ‘X.tail’
         imported from ‘Intro’ at test/BaseCompat.hs:7:1-17
         (and originally defined in ‘Data.List.NonEmpty’)
       ‘module X’ exports ‘X.tail’
         imported from ‘Prelude’ at test/BaseCompat.hs:61:1-178
         (and originally defined in ‘GHC.List’)
  |
4 |   module X
  |   ^^^^^^^^
```

**Comment by juhp** [comment]
To be on the safe side I will add an upper bound for now in Stackage Nightly.

**Comment by minad** [comment]
I am fixing it right now.

**Comment by minad** [comment]
Uploaded fixed 0.5.2.1 to hackage
